nndownload/nndownload.py

```diff
@@ -1911,23 +1912,186 @@ def download_thumbnail(session: requests.Session, filename: AnyStr, template_par
+            }
+        ],
+        "threads": [],
+    }
+
+    # Convert date limit to timestamp
+    last_time = get_niconico_timestamp()
+
+    # Process each comment thread
+    for thread in template_params["thread_params"]["targets"]:
+        thread_comments = fetch_comments_modern(
+            session,
+            template_params["id"],
+            template_params["comment_server"],
+            template_params["thread_key"],
+            thread,
+            template_params["thread_params"]["language"],
+            last_time,
+            comment_limit,
+        )
+
+        if thread_comments.get("globalCommentsData"):
+            if thread_comments["globalCommentsData"]["count"] > comments["globalComments"][0]["count"]:
+                comments["globalComments"][0]["count"] = thread_comments["globalCommentsData"]["count"]
+            comments["globalComments"][0]["id"] = thread_comments["globalCommentsData"]["id"]
+        comments["threads"].append(thread_comments["threadData"])
+
+    # Save comments to file
     with open(filename, "w", encoding="utf-8") as file:
-        json.dump(get_comments_request.json(), file, indent=4, ensure_ascii=False, sort_keys=True)
+        json.dump(comments, file, indent=4, ensure_ascii=False, sort_keys=True)
 
     output("Finished downloading comments for {0}.\n".format(template_params["id"]), logging.INFO)
 
+def fetch_comments_modern(
+    session: requests.Session,
+    video_id: str,
+    api_server: str,
+    thread_key: str,
+    thread: dict,
+    language: str,
+    last_time: int,
+    limit: int,
+) -> dict:
+    """Fetch comments using modern Niconico API (requires login)."""
+    global_comments_data = {
+        "count": 0,
+        "id": thread["id"],
+    }
+    thread_data = {
+        "commentCount": 0,
+        "comments": [],
+        "fork": thread["fork"],
+        "id": thread["id"]
+    }
+    base_data = {
+        "threadKey": thread_key,
+        "params": {
+            "language": language,
+            "targets": [thread]
+        }
+    }
+    fetched_count = 0
+    
+    while fetched_count < limit:
+        try:
+            payload = {
+                **base_data,
+                "additionals": {
+                    "res_from": -1000,
+                    "when": last_time
+                }
+            }
+            
+            headers = {
+                "content-type": "text/plain;charset=UTF-8",
+                "x-client-os-type": "others",
+                "x-frontend-id": "6",
+                "x-frontend-version": "0"
+            }
+            
+            response = session.post(
+                f"{api_server}/v1/threads",
+                json=payload,
+                headers=headers
+            )
+            data = response.json()
+            
+            # Handle errors
+            if "meta" in data and "errorCode" in data["meta"]:
+                handle_api_error(data["meta"]["errorCode"], session, video_id)
```

Comment:
I would rather keep this inside this method, instead of adding `handle_api_error`. If it makes sense we can genericize all the other API error handling to a method like this later.